OpenRA.Mods.Common/Traits/Pluggable.cs

```diff
@@ -23,19 +25,29 @@ public class PluggableInfo : ITraitInfo, UsesInit<PlugsInit>
 		[Desc("Conditions to grant for each accepted plug type.")]
 		public readonly Dictionary<string, string> Conditions = null;
 
+		[Desc("Conditions to grant for each accepted plug type.")]
+		public readonly Dictionary<string, BooleanExpression> Requirements = new Dictionary<string, BooleanExpression>();
+
 		[GrantedConditionReference]
 		public IEnumerable<string> LinterConditions { get { return Conditions.Values; } }
 
+		[ConsumedConditionReference]
+		public IEnumerable<string> ConsumedConditions
+		{
+			get { return Requirements.Values.SelectMany(r => r.Variables.Except(Conditions.Values)); }
```

Comment:
You're right, that is for the linter. I'll have to split this into `ConsumedConditions` and `LinterConsumedConditions` or is it valid to consider the same trait as a valid consumer of a condition?.